kolibri/plugins/learn/assets/src/state/actions/recommended.js

```diff
@@ -0,0 +1,280 @@
+import {
+  ContentNodeResource,
+  ContentNodeProgressResource,
+  SessionResource,
+  UserExamResource,
+  ExamLogResource,
+  ExamAttemptLogResource,
+} from 'kolibri.resources';
+
+import {
+  currentUserId,
+  isFacilityUser,
+  getCurrentChannelId,
+  getCurrentChannelObject,
+} from 'kolibri.coreVue.vuex.getters';
+
+import {
+  samePageCheckGenerator,
+  setChannelInfo,
+  handleApiError,
+} from 'kolibri.coreVue.vuex.actions';
+
+import { PageNames } from '../../constants';
+import { _contentState } from './main';
+
+import ConditionalPromise from 'kolibri.lib.conditionalPromise';
+import uniqBy from 'lodash/uniqBy';
+
+function _fetchPopular(channelId) {
+  const channelPayload = { channel_id: channelId };
+  const popularPayload = { popular: 'true' };
+  return ContentNodeResource.getCollection(channelPayload, popularPayload).fetch();
+}
+
+function _fetchNextSteps(channelId, state) {
+  const nextStepsPayload = { next_steps: currentUserId(state) };
+  const channelPayload = { channel_id: channelId };
+
+  if (isFacilityUser(state)) {
+    return ContentNodeResource.getCollection(channelPayload, nextStepsPayload).fetch();
+  }
+  return Promise.resolve([]);
+}
+
+function _fetchResume(channelId, state) {
+  const resumePayload = { resume: currentUserId(state) };
+  const channelPayload = { channel_id: channelId };
+
+  // all you really need is user_id. Should we check in parent?
+  if (isFacilityUser(state)) {
+    return ContentNodeResource.getCollection(channelPayload, resumePayload).fetch();
+  }
+  return Promise.resolve([]);
+}
+
+function _fetchOverview(channelId, cursor) {
+  const channelPayload = { channel_id: channelId };
+
+  return ContentNodeResource.getAllContentCollection(channelPayload, { cursor }).fetch();
+}
+
+function showLearnChannel(store, channelId, cursor) {
+  // Special case for when only the page number changes:
+  // Don't set the 'page loading' boolean, to prevent flash and loss of keyboard focus.
+  const state = store.state;
+  if (state.pageName !== PageNames.LEARN_CHANNEL || getCurrentChannelId(state) !== channelId) {
+    store.dispatch('CORE_SET_PAGE_LOADING', true);
+  }
+
+  // session should already be set by app.js
+  // const sessionPromise = SessionResource.getModel('current').fetch();
+  // do we really need to set the channel first?
+  // ConditionalPromise.all([sessionPromise, channelsPromise]).only(
+  const channelsPromise = setChannelInfo(store, channelId);
+  channelsPromise.only(
+    samePageCheckGenerator(store),
```

Comment:
Ah. Then it seems like the only cases I shouldn't be using `samePageCheckGenerator` is when a helper action is used across various `showPage` actions?